### PR TITLE
!fix: Drop old TFV version (< `v0.6.0`)  support

### LINKED ISSUE #117: Terraform Validator v0.6.0 change to release binary format not handled by Cloud Build builder Dockerfile
The `v0.6.0` release of Terraform Validator changes the format of the release binaries to a zipped tarball per these release notes : https://github.com/GoogleCloudPlatform/terraform-validator/releases/tag/v0.6.0

This also means the file names of the release binaries available in GCS have changed since version `0.5.1`. See the below example illustrating the difference :

- __v0.5.1__ : `gs://terraform-validator/releases/v0.5.1/terraform-validator-linux-amd64`
- __v0.6.0__ : `gs://terraform-validator/releases/v0.6.0/terraform-validator_linux_amd64-0.6.0.tar.gz`

This change is currently not supported by the [Cloud Build builder Dockerfile](https://github.com/terraform-google-modules/terraform-google-bootstrap/blob/master/modules/cloudbuild/cloudbuild_builder/Dockerfile) and results in a build error indicating it cannot find the terraform-validator binary.

Also the Dockerfile will need to be amended to unzip and extract the tarball as it only unzips it currently.

### PR DESCRIPTION
**Pull request by vovinacci**
As

- #121  is [not likely going to be worked on further](https://github.com/terraform-google-modules/terraform-google-bootstrap/pull/121#issuecomment-963351013)
- based on https://github.com/terraform-google-modules/terraform-google-bootstrap/pull/121#discussion_r704540217

This PR drops support for older TFV versions (< `v0.6.0`) with a breaking change.

Will require updating TFV version in https://github.com/terraform-google-modules/terraform-example-foundation/blob/0ee39ceff2dce8c3a3b61d59bb1d45a4d0687e74/0-bootstrap/main.tf#L120 along with module version update.

- Fixes #117 
- Closes #121

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/cloudbuild/cloudbuild_builder/Dockerfile [hunk 1]
```diff
@@ -32,7 +32,9 @@ RUN apt-get update && \
     sha256sum -c terraform_SHA256SUMS --status && \
     unzip terraform_linux_amd64.zip -d /builder/terraform && \
     rm -f terraform_linux_amd64.zip && \
-    gsutil cp gs://terraform-validator/releases/${ENV_TERRAFORM_VALIDATOR_RELEASE}/terraform-validator-linux-amd64 /builder/terraform/terraform-validator && \
+    TERRAFORM_VALIDATOR_VERSION=$(echo $ENV_TERRAFORM_VALIDATOR_RELEASE | sed -e 's/^v//') && \
+    TERRAFORM_VALIDATOR_GCS_PATH=gs://terraform-validator/releases/${ENV_TERRAFORM_VALIDATOR_RELEASE}/terraform-validator_linux_amd64-${TERRAFORM_VALIDATOR_VERSION}.tar.gz && \
+    gsutil cat "${TERRAFORM_VALIDATOR_GCS_PATH}" | tar -zxv -C /builder/terraform terraform-validator && \
```
Comment (bharathkkb) [inline modules/cloudbuild/cloudbuild_builder/Dockerfile:37]: Can we use `cp` than `cat`? It seems like cat does not validate checksum https://cloud.google.com/storage/docs/gsutil/commands/cat#:~:text=output%20to%20stdout.)-,warning,-%3A%20The%20gsutil%20cat

File: modules/cloudbuild/variables.tf [hunk 2]
```diff
@@ -146,7 +146,7 @@ variable "terraform_version_sha256sum" {
 variable "terraform_validator_release" {
   description = "Default terraform-validator release."
   type        = string
-  default     = "v0.4.0"
+  default     = "v0.6.0"
```
[the remark this answers is not in the document]
Comment (vovinacci) [reply modules/cloudbuild/variables.tf:149]: Done.

File: modules/cloudbuild/cloudbuild_builder/Dockerfile (hunk 1 above)
Comment (vovinacci) [reply modules/cloudbuild/cloudbuild_builder/Dockerfile]: Done.

Also updated Terraform to 0.13.7 - latest in 0.13. No real changes there, just [public key update](https://github.com/hashicorp/terraform/blob/v0.13/CHANGELOG.md)

Comment (vovinacci) [reply modules/cloudbuild/cloudbuild_builder/Dockerfile]: @bharathkkb PR review comments addressed.

**Review by bharathkkb — APPROVED** [review]
Thanks for working on this @vovinacci

**Pull request merged**